### ISSUE TITLE
Allow setting `log_dir` manually, `trainer.log_dir` is not found with `MLFlowLogger`

### ISSUE THREAD

**Issue by olipinski** [speediedan/finetuning-scheduler#16]
## 🐛 Bug

The current way of using `trainer.log_dir` does no work with loggers that do not user that variable, such as `MLFlowLogger` (https://github.com/Lightning-AI/pytorch-lightning/issues/20615)

### To Reproduce

```python
import torch
from finetuning_scheduler import FinetuningScheduler
from pytorch_lightning.loggers import MLFlowLogger
import lightning.pytorch as L
from torch.utils.data import DataLoader, Dataset

# Thanks to #20605 for the small classes!

class RandomDataset(Dataset):
    def __init__(self, size):
        self.data = torch.randn(size, 10)

    def __len__(self):
        return len(self.data)

    def __getitem__(self, idx):
        return self.data[idx], torch.tensor(0)  # Dummy target


class MinimalDataModule(L.LightningDataModule):
    def train_dataloader(self):
        return DataLoader(RandomDataset(100), batch_size=10)

    # when removing the val_dataloader method completely, the error is not raised
    def val_dataloader(self):
        return []


class MinimalModel(L.LightningModule):
    def __init__(self):
        super().__init__()
        self.linear = torch.nn.Linear(10, 1)

    def forward(self, x):
        return self.linear(x)

    def training_step(self, batch, batch_idx):
        x, y = batch
        loss = torch.nn.functional.mse_loss(self(x), y.float().unsqueeze(1))
        return loss

    def validation_step(self, batch, batch_idx):
        x, y = batch
        loss = torch.nn.functional.mse_loss(self(x), y.float().unsqueeze(1))
        return loss

    def configure_optimizers(self):
        return torch.optim.Adam(self.parameters(), lr=0.01)

mlflow_logger = MLFlowLogger(
    tracking_uri="sqlite:///test.db",
    experiment_name="test",
    run_name=f"ft-test",
)

ft_callback = FinetuningScheduler(gen_ft_sched_only=True)



# Initialise the trainer
trainer = L.Trainer(
    accelerator="auto",
    max_epochs=1,
    logger=[mlflow_logger],
    callbacks=[ft_callback],
    check_val_every_n_epoch=1,
)

model = MinimalModel()
data = MinimalDataModule()

trainer.fit(model, datamodule=data)
```

### Expected behavior

The fine-tuning scheduler uses a different log_dir, or allows us to set one.

### Environment

- Fine-Tuning Scheduler Version (e.g., 2.7.0): 2.5.0
- Lightning Version (e.g., 2.7.0): 2.5.0.post0
- PyTorch Version (e.g., 2.7.0): 2.6.0
- Python version (e.g., 3.12): 3.12.9
- OS (e.g., Linux): MacOS
- CUDA/cuDNN version: N/A
- GPU models and configuration: N/A
- How you installed PyTorch (`pip`, source): pip
- If compiling from source, the output of `torch.__config__.show()`:
- Any other relevant information:
